### PR TITLE
Representation Factory rebase

### PR DESCRIPTION
**Pull request by petersteneteg**

### CODE REVIEW

File: include/inviwo/core/common/inviwoapplication.h [hunk 1]
```diff
@@ -224,6 +226,8 @@ class IVW_CORE_API InviwoApplication : public Singleton<InviwoApplication>,
     auto dispatchFront(F&& f, Args&&... args)
         -> std::future<typename std::result_of<F(Args...)>::type>;
 
+    void dispatchFrontAndForget(std::function<void()> fun);
```
Comment (martinfalk) [inline include/inviwo/core/common/inviwoapplication.h:236]: what does this one do?

Comment (martinfalk) [reply include/inviwo/core/common/inviwoapplication.h:236]: what's forgotten?

File: include/inviwo/core/network/networklock.h [hunk 2]
```diff
@@ -47,8 +47,12 @@ struct IVW_CORE_API NetworkLock {
 
     NetworkLock(NetworkLock const&) = delete;
     NetworkLock& operator=(NetworkLock const& that) = delete;
-    NetworkLock(NetworkLock&& rhs) = delete;
-    NetworkLock& operator=(NetworkLock&& that) = delete;
+    NetworkLock(NetworkLock&& rhs) : network_(rhs.network_) { rhs.network_ = nullptr; };
+    NetworkLock& operator=(NetworkLock&& that) {
+        NetworkLock lock(std::move(that));
+        std::swap(network_, lock.network_);
```
Comment (martinfalk) [inline include/inviwo/core/network/networklock.h]: move implementation to cpp file?

File: src/core/datastructures/representationfactorymanager.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+#include <inviwo/core/datastructures/representationfactorymanager.h>
+
+namespace inviwo {
+
+RepresentationMetaFactory* RepresentationFactoryManager::representationMetaFactory_ = nullptr;
+RepresentationConverterMetaFactory* RepresentationFactoryManager::representationConverterMetaFactory_ = nullptr;
+
+
+    RepresentationFactoryManager::RepresentationFactoryManager()
+        : oldRepresentationMetaFactory_{representationMetaFactory_}
```
Comment (martinfalk) [inline src/core/datastructures/representationfactorymanager.cpp]: clang format

**Review by martinfalk — APPROVED** [review]

File: include/inviwo/core/common/inviwoapplication.h (hunk 1 above)
Comment (petersteneteg) [reply include/inviwo/core/common/inviwoapplication.h:236]: the return value...

**Pull request merged**